idaes/core/solvers/petsc.py

```diff
@@ -403,51 +413,71 @@ def petsc_dae_by_time_element(
             otherwise know the initial conditions.
         snes_options (dict): PETSc nonlinear equation solver options
         ts_options (dict): PETSc time-stepping solver options
-        wsl (bool): if True use WSL to run PETSc, if False don't use WSL to run
-            PETSc, if None automatic. The WSL is only for Windows.
         keepfiles (bool): pass to keepfiles arg for solvers
         symbolic_solver_labels (bool): pass to symbolic_solver_labels argument
             for solvers. If you want to read trajectory data from the
             time-stepping solver, this should be True.
-        vars_stub (str or None): Copy the `*.col` and `*.typ` files to the
-            working directory using this stub if not None.  These are needed to
-            interpret the trajectory data.
-        trajectory_save_prefix (str or None): If a string is provided the
-            trajectory data will be saved as gzipped json
-
-    Returns:
-        List of solver results objects from each solve. If there are initial
-        condition constraints and they are not skipped, the first object will
-        be from the initial condition solve.  Then there should be one for each
-        time element for each TS solve.
+        between (list or tuple): List of time points to integrate between. If
+            None use all time points in the model. Generally the list should
+            start with the first time point and end with the last time point;
+            however, this is not a requirement and the are situations where you
+            may not want to include them. If you are not including the first
+            time point, you should take extra care with the initial conditions.
+            If the initial conditions are already correct consider using the
+            ``skip_initial`` option if the first time point is not included.
+        interpolate (bool): if True and trajectory is read, interpolate model
+            values from the trajectory
+        calculate_derivaties: (bool) if True, calculate the derivative values
+            based on the values of the differential variables in the discretized
+            Pyomo model.
+
+    Returns (PetscDAEResults):
+        See PetscDAEResults documentation for more informations.
     """
+    tj = None
+    if between is None:
+        between = time
+    else:
+        for t in between:
+            try:
+                assert t in time
+            except AssertionError:
+                raise RuntimeError("Elements of between must be in the time set")
+        between = pyo.Set(initialize=sorted(between))
+        between.construct()
```

Comment:
I would recommend just working with the sorted Python list of values instead of creating a floating Pyomo `Set` component that isn't attached to a model.

Reply:
I think the reason I did this was to use first(), last(), and at(), which I was using for the full time set.  I guess alternatively, I could us a list here are make a list from the full time set in the other if-branch.